File: 프로그래머스/LV2/괄호 회전하기.py
def solution(s):
    answer = 0
    s = list(s)
    for i in range(len(s)):
        stack = []
        for j in range(len(s)):
            if len(stack) > 0:
                if stack[-1] == "[" and s[j] == "]":
                    stack.pop()
                elif stack[-1] == "{" and s[j] == "}":
                    stack.pop()
                elif stack[-1] == "(" and s[j] == ")":
                    stack.pop()
                else:
                    stack.append(s[j])
            else:
                stack.append(s[j])
        if len(stack) == 0:
            answer += 1
        s.append(s.pop(0))

    return answer

File: 프로그래머스/javascript/LV2/괄호 회전하기.js
// 올바른 괄호 문자열인지 확인
function isCorrect(str) {
  let stack = [];
  for (const a of str) {
    if (a === "[" || a === "{" || a === "(") {
      stack.push(a);
    } else {
      if (a === "]" && stack[stack.length - 1] === "[") {
        stack.pop();
      } else if (a === "}" && stack[stack.length - 1] === "{") {
        stack.pop();
      } else if (a === ")" && stack[stack.length - 1] === "(") {
        stack.pop();
      }
    }
  }

  return stack.length === 0;
}

function solution(s) {
  let answer = 0;
  if (s.length % 2) return 0;

  for (let i = 0; i < s.length; i++) {
    let str = s.slice(i) + s.slice(0, i);
    if (isCorrect(str)) answer++;
  }

  return answer;
}
